File: Freedom/微信/通讯录/WXTagsViewController.m
//  TLTagsViewController.m
//  Freedom
// Created by Super
#import "WXTagsViewController.h"
#import "WXTableViewCell.h"
#import "WXUserHelper.h"
@interface WXTagCell : WXTableViewCell
@property (nonatomic, strong) NSString *title;
@end
@interface WXTagCell()
@property (nonatomic, strong) UILabel *titleLabel;
@end
@implementation WXTagCell
- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        self.leftSeparatorSpace = 15;
        [self.contentView addSubview:self.titleLabel];
        
        [self p_addMasonry];
    }
    return self;
}
- (void)setTitle:(NSString *)title{
    _title = title;
    [self.titleLabel setText:title];
}
#pragma mark - 
- (void)p_addMasonry{
    [self.titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.mas_equalTo(self.leftSeparatorSpace);
        make.right.mas_lessThanOrEqualTo(-self.leftSeparatorSpace);
        make.centerY.mas_equalTo(self.contentView);
    }];
}
#pragma mark - 
- (UILabel *)titleLabel{
    if (_titleLabel == nil) {
        _titleLabel = [[UILabel alloc] init];
        [_titleLabel setFont:[UIFont systemFontOfSize:15.0f]];
    }
    return _titleLabel;
}
@end
@implementation WXTagsViewController
- (void)viewDidLoad{
    [super viewDidLoad];
    [self.navigationItem setTitle:@"标签"];
    [self.view setBackgroundColor:[UIColor whiteColor]];
    
    [self registerCellClass];
    
    UIBarButtonItem *rightBarButton = [[UIBarButtonItem alloc] initWithTitle:@"新建" style:UIBarButtonItemStylePlain target:self action:@selector(rightBarButtonDown:)];
    [self.navigationItem setRightBarButtonItem:rightBarButton];
    
    self.data = [WXFriendHelper sharedFriendHelper].tagsData;
}
#pragma mark - Event Response -
- (void)rightBarButtonDown:(UIBarButtonItem *)sender{
    
}
#pragma mark - Public Methods -
- (void)registerCellClass{
    [self.tableView registerClass:[WXTagCell class] forCellReuseIdentifier:@"TLTagCell"];
}
#pragma mark - Delegate -
//MARK: UITableViewDataSource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.data.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    WXUserGroup *group = [self.data objectAtIndex:indexPath.row];
    WXTagCell *cell = [tableView dequeueReusableCellWithIdentifier:@"TLTagCell"];
    [cell setTitle:[NSString stringWithFormat:@"%@(%ld)", group.groupName, (long)group.count]];
    [cell setTopLineStyle:(indexPath.row == 0 ? TLCellLineStyleFill : TLCellLineStyleNone)];
    [cell setBottomLineStyle:(indexPath.row == self.data.count - 1 ? TLCellLineStyleFill : TLCellLineStyleDefault)];
    return cell;
}
//MARK: UITableViewDelegate
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return 55.0f;
}
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    [tableView deselectRowAtIndexPath:indexPath animated:NO];
}
@end

File: Freedom/微信/通讯录/WXTagsViewController.swift
//
//  WXTagsViewController.swift
//  Freedom
import SnapKit
import XExtension
import Foundation
class WXTagCell: BaseTableViewCell<Any> {
    required init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        title = UILabel()
        title.font = UIFont.systemFont(ofSize: 15.0)
        contentView.addSubview(title)
    }
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
class WXTagsViewController: BaseTableViewController {
    var data: [WXUserGroup] = []
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "标签"
        view.backgroundColor = UIColor.white
        tableView.register(WXTagCell.self, forCellReuseIdentifier: WXTagCell.identifier)
        let rightBarButton = UIBarButtonItem(title: "新建", style: .plain, target: self, action: #selector(WXTagsViewController.rightBarButtonDown(_:)))
        navigationItem.rightBarButtonItem = rightBarButton
        data = WXFriendHelper.shared.tagsData
    }
    @objc func rightBarButtonDown(_ sender: UIBarButtonItem) {
    }
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return data.count
    }
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let group = data[indexPath.row]
        let cell = tableView.dequeueCell(WXTagCell.self)
        cell.title.text = String(format: "%@(%ld)", group.groupName, Int(group.users.count))
        return cell
    }
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 55.0
    }
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: false)
    }

}
